src/modules/collections/services/collections.service.ts

```diff
@@ -184,22 +185,48 @@ export class CollectionsService {
 		return response?.data?.delete_users_collection?.affected_rows || 0;
 	}
 
+	public async findObjectInCollection(
+		collectionId: string,
+		objectId: string
+	): Promise<IeObject | null> {
+		const response = await this.dataService.execute(FIND_OBJECT_IN_COLLECTION, {
+			collectionId,
+			objectId,
+		});
+		const foundObject = response?.data?.users_collection_ie?.[0];
+		this.logger.debug(`Found object ${objectId} in ${collectionId}`);
+
+		return this.adaptCollectionObjectLink(foundObject);
+	}
+
 	public async addObjectToCollection(collectionId: string, objectId: string): Promise<IeObject> {
+		const collectionObject = await this.findObjectInCollection(collectionId, objectId);
```

Comment:
Kunnen we dit niet opvangen met die on_conflict in graphql? (zoals een insert ignore in SQL), of is dit nodig voor de FE?

Reply:
Het was vooral om naar de frontend toe een mooie error te voorzien. Anders is het een generieke error, waar we moeilijk een mooi error bericht voor kunnen tonen.